exist-core/src/main/java/org/exist/http/RESTServer.java

```diff
@@ -537,9 +537,9 @@ public void doGet(final DBBroker broker, final Txn transaction, final HttpServle
                         LOG.debug(e.getMessage(), e);
                     }
                     if (MimeType.XML_TYPE.getName().equals(mimeType)) {
-                        writeXPathException(response, HttpServletResponse.SC_BAD_REQUEST, encoding, query, path, e);
+                        writeXPathException(response, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, encoding, query, path, e);
```

Comment:
is it an system error or a user error?

Reply:
@dizzzz I have only changed the codes where a user is making a request to execute a query that is already stored in the database. I think these could be classed as Internal Server Errors.

The requests that send a query to eXist-db, I have left as Bad Request error.